src/backend/Level0/CHIPBackendLevel0.cc

```diff
@@ -215,6 +211,28 @@ createSampler(CHIPDeviceLevel0 *ChipDev, const hipResourceDesc *PResDesc,
 // CHIPEventLevel0
 // ***********************************************************************
 
+void CHIPEventLevel0::associateCmdList(CHIPContextLevel0 *ChipContext,
+                                       ze_command_list_handle_t CmdList) {
+  logTrace("CHIPEventLevel0({})::associateCmdList({})", (void *)this,
+           (void *)CmdList);
+  assert(AssocCmdList_ == nullptr && "command list already associated!");
+  assert(AssocContext_ == nullptr && "queue already associated!");
+  AssocCmdList_ = CmdList;
+  AssocContext_ = ChipContext;
+}
+
+void CHIPEventLevel0::disassociateCmdList() {
+  assert(AssocCmdList_ != nullptr && "command list not associated!");
+  assert(AssocContext_ != nullptr && "queue not associated!");
+  logTrace("CHIPEventLevel0({})::disassociateCmdList({})", (void *)this,
+           (void *)AssocCmdList_);
+  auto Status = zeCommandListReset(AssocCmdList_);
+  CHIPERR_CHECK_LOG_AND_THROW(Status, ZE_RESULT_SUCCESS, hipErrorTbd);
+  AssocContext_->returnCmdList(AssocCmdList_);
+  AssocCmdList_ = nullptr;
+  AssocContext_ = nullptr;
```

Comment:
This looks like a logic that should be in CHIPEventLevel0::reset(). Now, it looks like there is event reset logic in two places.

Reply:
This does not reset the event, rather only deals with release the associated command list when this event fires. Event can fire and then still remain unreset. If we go with a struct for containing command lists then ```  auto Status = zeCommandListReset(AssocCmdList_);
  CHIPERR_CHECK_LOG_AND_THROW(Status, ZE_RESULT_SUCCESS, hipErrorTbd);``` could go into the `CmdList.reset()`` and that's how I had it in my initial draft but decided to keep it as simple as possible for now.

Reply:
>This does not reset the event, rather only deals with release the associated command list when this event fires. Event can fire and then still remain unreset.

The events associated to command lists, which are disassociated in the event garbage collector, are going to be returned event pool soon and eventually to be reset, right? What are the issues that prevent the command list disassociate logic to live in CHIPEventLevel0::reset()?

Reply:
> are going to be returned event pool soon and eventually to be reset, right? 

not necessarily right away. An event could fire (at which point we don't need the command list and can recycle it, thus disassociate it) but still remain refcounted somewhere for example in a dependency chain somewhere.